src/api/nft-namespace.ts

```diff
@@ -0,0 +1,646 @@
+   * Alchemy marks spam contracts, go to
+   * https://docs.alchemy.com/alchemy/enhanced-apis/nft-api/nft-api-faq#nft-spam-classification.
+   *
+   * @beta
+   */
+  getSpamContracts(): Promise<string[]> {
+    return requestHttpWithBackoff<undefined, string[]>(
+      this.config,
+      AlchemyApiType.NFT,
+      'getSpamContracts',
+      undefined
+    );
+  }
+
+  /**
+   * Returns the floor prices of a NFT contract by marketplace.
+   *
+   * @param contractAddress - The contract address for the NFT collection.
+   * @beta
+   */
+  getFloorPrice(contractAddress: string): Promise<GetFloorPriceResponse> {
+    return requestHttpWithBackoff<GetFloorPriceParams, GetFloorPriceResponse>(
+      this.config,
+      AlchemyApiType.NFT,
+      'getFloorPrice',
+      {
+        contractAddress
+      }
+    );
+  }
+
+  /**
+   * Refreshes the cached metadata for a provided NFT contract address and token
+   * id. Returns a boolean value indicating whether the metadata was refreshed.
+   *
+   * This method is useful when you want to refresh the metadata for a NFT that
+   * has been updated since the last time it was fetched. Note that the backend
+   * only allows one refresh per token every 15 minutes, globally for all users.
+   * The last refresh time for an NFT can be accessed on the
+   * {@link Nft.timeLastUpdated} field.
+   *
+   * To trigger a refresh for all NFTs in a contract, use {@link refreshContract} instead.
+   *
+   * @param contractAddress - The contract address of the NFT.
+   * @param tokenId - The token id of the NFT.
+   */
+  async refreshNftMetadata(
+    contractAddress: string,
+    tokenId: BigNumberish
+  ): Promise<boolean> {
+    const tokenIdString = BigNumber.from(tokenId!).toString();
+    const first = await this.getNftMetadata(contractAddress, tokenIdString);
+    const second = await refresh(this.config, contractAddress, tokenIdString);
+    return first.timeLastUpdated !== second.timeLastUpdated;
+  }
+
+  /**
+   * Triggers a metadata refresh all NFTs in the provided contract address. This
+   * method is useful after an NFT collection is revealed.
+   *
+   * Refreshes are queued on the Alchemy backend and may take time to fully
+   * process. To refresh the metadata for a specific token, use the
+   * {@link refreshNftMetadata} method instead.
+   *
+   * @param contractAddress - The contract address of the NFT collection.
+   * @beta
+   */
+  async refreshContract(
+    contractAddress: string
+  ): Promise<RefreshContractResult> {
+    const response = await requestHttpWithBackoff<
+      ReingestContractParams,
+      RawReingestContractResponse
+    >(this.config, AlchemyApiType.NFT, 'reingestContract', {
+      contractAddress
+    });
+
+    return {
+      contractAddress: response.contractAddress,
+      refreshState: parseReingestionState(response.reingestionState),
+      progress: response.progress
+    };
+  }
+}
+
+async function refresh(
```

Comment:
Should we make this private? Assuming it's an internal method

Reply:
The function isn't exported and isn't a member function on the namespace class, so we're fine.